### PR TITLE
Use Hamcrest matchers and assertThat() in ReindexRenamedSettingTests

### PR DESCRIPTION
**Pull request by tlfeng**
### Description

- Flip the `assertEquals()` arguments in `ReindexRenamedSettingTests` class

According to the comment in https://github.com/opensearch-project/OpenSearch/pull/2463#discussion_r827424263:

> The first argument of the `assert` is the _expected_ value, and the second is the _actual_ value.

The unit test class `ReindexRenamedSettingTests` were added in PR https://github.com/opensearch-project/OpenSearch/pull/2221 / commit https://github.com/opensearch-project/OpenSearch/commit/63c75d1b1d188a4bc3917169001aa8711ad702ef

### Issues Resolved
Related to issue https://github.com/opensearch-project/OpenSearch/issues/1547
 
### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by VachaShah — APPROVED** [review]

**Comment by tlfeng** [comment]
in log 3509:
```
> Task :server:internalClusterTest

REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.search.SearchCancellationIT.testMSearchChildRequestCancellationWithClusterLevelTimeout" -Dtests.seed=E7E590B85679E426 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=no -Dtests.timezone=Australia/South -Druntime.java=17

org.opensearch.search.SearchCancellationIT > testMSearchChildRequestCancellationWithClusterLevelTimeout FAILED
    java.lang.AssertionError: Actual child request with cancellation failure is different that expected expected:<[0, 1]> but was:<[]>
        at __randomizedtesting.SeedInfo.seed([E7E590B85679E426:59585B5DC76F0451]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:120)
        at org.opensearch.search.SearchCancellationIT.ensureMSearchWasCancelled(SearchCancellationIT.java:194)
        at org.opensearch.search.SearchCancellationIT.testMSearchChildRequestCancellationWithClusterLevelTimeout(SearchCancellationIT.java:537)
```
It's reported in issue https://github.com/opensearch-project/OpenSearch/issues/2242

**Review by owaiskazi19 — APPROVED** [review]

**Review by andrross — APPROVED** [review]
> The first argument of the assert is the expected value, and the second is the actual value.

FYI, this is one reason to prefer the Hamcrest matcher style of `assertThat(actualValue, equalTo(expectedValue))` because it reads more naturally and you don't have know the ordering.

**Comment by tlfeng** [comment]
@andrross Thank you for sharing the knowledge! Now I know `assertThat()` should be preferred to use.
> Hamcrest strives to make your tests as readable as possible.

p.s. the advantages of `assertThat()` are shown in the release notes of junit 4.4: http://junit.sourceforge.net/doc/ReleaseNotes4.4.html

**Comment by tlfeng** [comment]
In Log 3518:
```
> Task :qa:remote-clusters:integTest FAILED

REPRODUCE WITH: ./gradlew ':qa:remote-clusters:integTest' --tests "org.opensearch.cluster.remote.test.RemoteClustersIT.testHAProxyModeConnectionWorks" -Dtests.seed=DB46B9E87253F055 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=no -Dtests.timezone=America/Argentina/San_Juan -Druntime.java=17

WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by org.gradle.api.internal.tasks.testing.worker.TestWorker (file:/home/ubuntu/.gradle/wrapper/dists/gradle-7.3.3-all/4295vidhdd9hd3gbjyw1xqxpo/gradle-7.3.3/lib/plugins/gradle-testing-base-7.3.3.jar)
WARNING: Please consider reporting this to the maintainers of org.gradle.api.internal.tasks.testing.worker.TestWorker
WARNING: System::setSecurityManager will be removed in a future release
org.opensearch.cluster.remote.test.RemoteClustersIT > testHAProxyModeConnectionWorks FAILED
    java.lang.AssertionError
        at __randomizedtesting.SeedInfo.seed([DB46B9E87253F055:DC306C2349D58608]:0)
        at org.junit.Assert.fail(Assert.java:87)
        at org.junit.Assert.assertTrue(Assert.java:42)
        at org.junit.Assert.assertTrue(Assert.java:53)
        at org.opensearch.cluster.remote.test.RemoteClustersIT.testHAProxyModeConnectionWorks(RemoteClustersIT.java:125)
```

It's reported in https://github.com/opensearch-project/OpenSearch/issues/1703, this one fails very often recently.

File: modules/reindex/src/test/java/org/opensearch/index/reindex/ReindexRenamedSettingTests.java [hunk 1]
```diff
@@ -8,13 +8,16 @@
...
 
+import org.hamcrest.MatcherAssert;
 import org.opensearch.common.settings.Setting;
 import org.opensearch.common.settings.Settings;
 import org.opensearch.test.OpenSearchTestCase;
 
 import java.util.Arrays;
 import java.util.List;
 
+import static org.hamcrest.Matchers.equalTo;
```
Comment (tlfeng) [inline modules/reindex/src/test/java/org/opensearch/index/reindex/ReindexRenamedSettingTests.java:19]: According to suggestion from @andrross , I updated all the assertions in this file to use APIs from Hamcrest. I will update the PR description later.
although `assertThat()` is also available in junit, but it's deprecated. (https://junit.org/junit4/javadoc/4.13/org/junit/Assert.html#assertThat(java.lang.String,%20T,%20org.hamcrest.Matcher))

**Comment by tlfeng** [comment]
In log 3522:
```
REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.gateway.RecoveryFromGatewayIT.testReuseInFileBasedPeerRecovery" -Dtests.seed=678B15424FD7EBE5 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=sr-RS -Dtests.timezone=Pacific/Midway -Druntime.java=17

org.opensearch.gateway.RecoveryFromGatewayIT > testReuseInFileBasedPeerRecovery FAILED
    java.lang.AssertionError: shard [test][0] on node [node_t1] has pending operations:
     --> RetentionLeaseBackgroundSyncAction.Request{retentionLeases=RetentionLeases{primaryTerm=1, version=1480, leases={peer_recovery/awjm15faRpeBMJiOvn_QMQ=RetentionLease{id='peer_recovery/awjm15faRpeBMJiOvn_QMQ', retainingSequenceNumber=786, timestamp=1647629686105, source='peer recovery'}, peer_recovery/7LnHvaeSRV6f3UsOaIJARg=RetentionLease{id='peer_recovery/7LnHvaeSRV6f3UsOaIJARg', retainingSequenceNumber=786, timestamp=1647629686105, source='peer recovery'}}}, shardId=[test][0], timeout=1m, index='test', waitForActiveShards=0}
    	at org.opensearch.index.shard.IndexShardOperationPermits.acquire(IndexShardOperationPermits.java:248)
    	at org.opensearch.index.shard.IndexShard.acquirePrimaryOperationPermit(IndexShard.java:3146)
```
It's reported in issue https://github.com/opensearch-project/OpenSearch/issues/1746, this one also fails frequently recently.

**Comment by tlfeng** [comment]
In report 3535:
failed test:
org.opensearch.index.ShardIndexingPressureConcurrentExecutionTests.html#testCoordinatingPrimaryThreadedUpdateToShardLimitsAndRejections

start gradle check

**Pull request merged**